salt/metalk8s/saltmaster/deployed.sls

```diff
@@ -0,0 +1,31 @@
+{% set salt_master_image = 'salt-master' %}
+{% set salt_master_version = '2018.3.3-1' %}
+{% set version = 'dev' %}
+
+Create salt master directories:
+  file.directory:
+    - user: root
+    - group: root
+    - mode: '0700'
+    - makedirs: true
+    - names:
+      - /etc/salt
+      - /var/cache/salt
+      - /var/run/salt
+
+Install salt master manifest:
+  file.managed:
+    - name: /etc/kubernetes/manifests/salt-master-pod.yaml
+    - source: salt://metalk8s/saltmaster/files/salt-master-pod.yaml.j2
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: '0644'
+    - makedirs: false
+    - backup: false
+    - require:
```

Comment:
To be discussed:

Strictly speaking, there's no hard requirement: this file *can* be created when those other directories don't exist. But, when `kubelet` sees the manifest file, it will fail to actually start the `salt-master` `Pod` because they don't exist, and say so in logs.

What are the exact semantics of `require` we should use? @scality/metalk8s

Reply:
Just change the name of the state accordingly. If the state is "Install and start salt master", then it's perfectly OK to have this requirement.

Reply:
@sayf-eddine-scality is this solved?

Reply:
most of them are solved but not pushed everything will make sens once i repush